src/lib/openjp2/j2k.c

```diff
@@ -2276,6 +2305,122 @@ static OPJ_BOOL opj_j2k_read_siz(opj_j2k_t *p_j2k,
             l_cp->m_specific_param.m_dec.m_reduce; /* reducing factor per component */
         ++l_img_comp;
     }
+    if (!p_j2k->dump_state) {
```

Comment:
As discussed in another comment, I don't think those checks should be put in the library code